Core/include/AnalysisCore/TrackSelectorPID.h

```diff
@@ -477,6 +476,190 @@ class TrackSelectorPID
     return isSelRICH || isSelTOF;
   }
 
+  ////////////////////////////////new sourav: is pion/////////////////////////////////
+  template <typename T>
+  bool isPion(const T& track, int usedetector = 0)
+  {
+    bool hasRICH = track.richId() > -1;
+    //bool hasTOF = track.tofId() > -1;
+    bool hasTOF = isValidTrackPIDTOF(track);
+
+    auto nSigmaTOFEl = track.tofNSigmaEl();
+    auto nSigmaTOFPi = track.tofNSigmaPi();
+    auto nSigmaTOFKa = track.tofNSigmaKa();
+
+    /*auto nSigmaTOFEl = hasTOF ? track.tof().tofNsigmaEl() : -1000.;
+    auto nSigmaTOFPi = hasTOF ? track.tof().richNsigmaPi() : -1000.;
+    auto nSigmaTOFKa = hasTOF ? track.tof().richNsigmaKa() : -1000.;
+     */
+    auto nSigmaRICHEl = hasRICH ? track.rich().richNsigmaEl() : -1000.;
+    auto nSigmaRICHPi = hasRICH ? track.rich().richNsigmaPi() : -1000.;
+    auto nSigmaRICHKa = hasRICH ? track.rich().richNsigmaKa() : -1000.;
+
+    auto p = track.p();
+    bool ispionselected = false;
+
+    if (!hasTOF) {
+      return ispionselected;
+    }
+
+    if (usedetector == 0) {
+      if (std::abs(nSigmaTOFPi) < mNSigmaTOFMax) {
+        ispionselected = true;
+      } else {
```

Comment:
Use `switch` for `usedetector`.
Why do you need `ispionselected = true` instead of `return true`?